grails-app/services/streama/TheMovieDbService.groovy

```diff
@@ -20,8 +20,16 @@ class TheMovieDbService {
 
 
   def getSimilarMovies(movieId){
-    def JsonContent = new URL(BASE_URL + "/movie/$movieId/similar?api_key=$API_KEY").text
-    return new JsonSlurper().parseText(JsonContent)
+    def JsonContent_similar = new URL(BASE_URL + "/movie/$movieId/similar?api_key=$API_KEY").text
+    def json_similar = new JsonSlurper().parseText(JsonContent_similar)
+    for(item in json_similar.results ){
+      def JsonContent_trailer = new URL(BASE_URL + "/movie/$item.id/videos?api_key=$API_KEY").text
+      def json_trailer = new JsonSlurper().parseText(JsonContent_trailer)
+      item.mediatype = "Movie"
+      item.trailerKey = json_trailer.results[0].key
```

Comment:
again, u gotta be null-safe here. json_trailer or reults[0] might be null. use `json_trailer?.results[0]?.key`